src/sdk/pynni/nni/_graph_utils.py

```diff
@@ -510,6 +518,65 @@ def _build_index(self, nodes_op):
                 output_to_node[output] = node
         return name_to_node, input_to_node, output_to_node
 
+    def _is_key_func(self, node_cpp):
+        """
+        Judge if a cpp node is a key function node.
+        If so, we should not merge this node into the
+        adjacent node.
+        """
+        if node_cpp.kind().startswith('aten::'):
+            # the nodes that start with 'aten' are key function
+            # nodes
+            return True
+        if node_cpp.kind() in [LIST_UNPACK_KIND, TUPLE_UNPACK_KIND]:
+            # We cannot merge the List/Tuple
+            # Construct/Unpack func into other nodes, else it
+            # may lead to a graph construction error.
+            return True
+        return False
+
+    def unpack_manually(self):
+        """
+        Unpack the tensor tuple or tensor list manually,
+        and remove the ListUnpack/TupleUnpack node from
+        the graph. Note: this function will change the
+        graph structure.
+        """
+        if hasattr(self, 'unpacked'):
+            # if already unpacked the tuple/list manually
+            return
+        for node in self.nodes_py.nodes_op:
+            if node.op_type in [TUPLE_UNPACK_KIND, LIST_UNPACK_KIND]:
+                unpack_cpp = node.key_node
+                last_cpp = list(unpack_cpp.inputs())[0].node()
```

Comment:
index is 0, why call it `last`?

Reply:
The `last` node actually refers to the previous(last) visited node.  In most scenarios, this `last_cpp` is the corresponding construct node of the tuple/list.